index/index.go

```diff
@@ -337,34 +360,76 @@ func (idx *Indexer) loopIteration(
 }
 
 func (idx *Indexer) ensureIndex(
-	ctx context.Context, indexType string, docType string,
+	ctx context.Context, indexType string, docType string, lang string,
 ) (string, error) {
-	name := fmt.Sprintf("%s-%s-%s",
-		indexType, idx.name, nonAlphaNum.ReplaceAllString(docType, "_"))
+	safeDocType := nonAlphaNum.ReplaceAllString(docType, "_")
+
+	config, err := GetLanguageConfig(lang)
+	if err != nil {
+		return "", fmt.Errorf("could not get language config: %w", err)
+	}
+
+	index := fmt.Sprintf("%s-%s-%s-%s", indexType, idx.name, safeDocType, config.Name)
```

Comment:
When we have a lot of these `%s-%s-%s...` I think that the intent and relationships between the different names become a bit unclear.

Something like this would make it clearer to me:
```go
	indexTypeRoot := fmt.Sprintf("%s-%s-%s", indexType, idx.name, safeDocType)
	index := fmt.Sprintf("%s-%s", indexTypeRoot, lang)

	aliases := []string{indexTypeRoot}

	// The language of the document might or might not be qualified with a
	// region specifier (en-GB vs en), check if we need to add a language
	// alias.
	if lang != config.Language {
		aliases = append(aliases, fmt.Sprintf(
			"%s-%s", indexTypeRoot, config.Language))
	}
```

The `config.Name` here is a bit unclear to me as well, and can be completely dropped if we normalise language early.

Reply:
If aliases are an array we can ensure them like this:
```go
	for _, alias := range aliases {
		err = idx.ensureAlias(index, alias)
		if err != nil {
			return "", fmt.Errorf("could not ensure alias %q: %w", alias, err)
		}
	}
```

Reply:
While working on this I realized that we still need a region name for indices that are not qualified with a region specifier, as there will be a name collision with the language alias. We can either use a default region name as fallback, or settle on another naming scheme for aliases. 

I took the first approach, so given a language code of `sv` we will create the index
```
percolate-v1-core_article-sv-unspecified
```
and set up the aliases 
```
percolate-v1-core_article
percolate-v1-core_article-sv
```

Reply:
Ah, I assumed that it was fine for aliases to have naming collisions with indexes, good catch.